core/server/common/src/main/java/alluxio/master/journal/raft/transport/CopycatGrpcTransport.java

```diff
@@ -75,14 +81,16 @@ public CopycatGrpcTransport(AlluxioConfiguration clientConf, AlluxioConfiguratio
 
     mClients = new LinkedList<>();
     mServers = new LinkedList<>();
+    mExecutor = Executors
+        .newCachedThreadPool(ThreadFactoryUtils.build("copycat-transport-worker-%d", true));
```

Comment:
Is cached threadpool the best choice, and can we share the executor between the client and server?

Reply:
Since connection building/termination is only required at certain times such as boot-up/reconfigure/terminate, we want a thread pool that doesn't consume resources while we are not in those stages. So cached thread pool with a 0 core pool size is preferred. We could also put a max count to thread pool size by creating ThreadPoolExecutor directly. However I don't see how that would be necessary.